sourcegraph/context.go

```diff
@@ -150,14 +153,22 @@ func (contextCredentials) GetRequestMetadata(ctx context.Context, uri ...string)
 		if m == nil {
 			m = credMD
 		} else {
+			// Copy the map to avoid a data race writing to the map returned by
+			// clientMetaDataFromContext.
+			cpy := make(map[string]string)
```

Comment:
Since you know the size that `cpy` will be (same as `m`), you could, if desired, give `make` the optional capacity hint parameter.